pkg/migrate.go

```diff
@@ -1,48 +1,95 @@
 package pkg
 
 import (
+	"errors"
+	"fmt"
 	"github.com/nuts-foundation/nuts-crypto/pkg/types"
 	core "github.com/nuts-foundation/nuts-go-core"
 	"github.com/nuts-foundation/nuts-registry/pkg/db"
+	"github.com/nuts-foundation/nuts-registry/pkg/events"
+	errors2 "github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-// verifyAndMigrateRegistry verifies the data in the registry, migrating data whenever required (e.g. issue missing certificates).
-// When a failed verification cannot be migrated, an error is returned.
-func (r *Registry) verifyAndMigrateRegistry(config core.NutsConfigValues) {
-	r.logger().Info("Verifying registry integrity...")
+// verify verifies the data in the registry, migrating data whenever required (e.g. issue missing certificates) when autoFix=true.
+// The events that result from fixing tge data are returned.
+func (r *Registry) verify(config core.NutsConfigValues, autoFix bool) ([]events.Event, bool, error) {
+	r.logger().Infof("Verifying registry integrity (autofix issues=%v)...", autoFix)
+	resultingEvents := make([]events.Event, 0)
 	// Assert vendor is registered
 	identity := config.Identity()
 	vendor := r.Db.VendorByID(identity)
+	fixRequired := false
+	var event events.Event
+	var err error
 	if vendor == nil {
-		logrus.Warnf("Configured vendor (%s) is not registered, please register it using the 'register-vendor' CLI command.", identity)
+		r.logger().Warnf("Configured vendor (%s) is not registered, please register it using the 'register-vendor' CLI command.", identity)
```

Comment:
Shouldn't these be errors -> so they can be propagated to the CLI or API client?

Reply:
This one, maybe, yes. But the rest not really since it's just information on what is found during verification and fixups. That information is useful to the CLI/API client too, though. Maybe we should pipe the log output while executing an API command to the client? Although maybe that's true for all CLI commands (that you want the log output piped to the CLI so it can be inspected by the administrator).

I'll make this one warning an error.